test/extensions/filters/http/ext_proc/ext_proc_grpc_fuzz_helper.cc

```diff
@@ -0,0 +1,657 @@
+#include "test/extensions/filters/http/ext_proc/ext_proc_grpc_fuzz_helper.h"
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/extensions/filters/http/ext_proc/v3alpha/ext_proc.pb.h"
+#include "envoy/service/ext_proc/v3alpha/external_processor.pb.h"
+#include "envoy/type/v3/http_status.pb.h"
+
+#include "source/common/common/thread.h"
+
+#include "test/common/http/common.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+using envoy::config::core::v3::HeaderValue;
+using envoy::config::core::v3::HeaderValueOption;
+using envoy::extensions::filters::http::ext_proc::v3alpha::ProcessingMode;
+using envoy::service::ext_proc::v3alpha::BodyResponse;
+using envoy::service::ext_proc::v3alpha::CommonResponse;
+using envoy::service::ext_proc::v3alpha::HeaderMutation;
+using envoy::service::ext_proc::v3alpha::HeadersResponse;
+using envoy::service::ext_proc::v3alpha::HttpBody;
+using envoy::service::ext_proc::v3alpha::HttpHeaders;
+using envoy::service::ext_proc::v3alpha::HttpTrailers;
+using envoy::service::ext_proc::v3alpha::ImmediateResponse;
+using envoy::service::ext_proc::v3alpha::ProcessingRequest;
+using envoy::service::ext_proc::v3alpha::ProcessingResponse;
+using envoy::service::ext_proc::v3alpha::TrailersResponse;
+using envoy::type::v3::StatusCode;
+
+ExtProcFuzzHelper::ExtProcFuzzHelper(FuzzedDataProvider* provider) {
+  provider_ = provider;
+  immediate_resp_sent_ = false;
+}
+
+std::string ExtProcFuzzHelper::consumeRepeatedString() {
+  uint32_t str_len = provider_->ConsumeIntegralInRange<uint32_t>(0, ext_proc_fuzz_max_data_size);
+  return std::string(str_len, 'b');
+}
+
+// TODO(ikepolinsky): should this function be put in a standard place?
+// Since FuzzedDataProvider requires enums to define a kMaxValue, we cannot
+// use the envoy::type::v3::StatusCode enum directly. Additionally this allows
+// us to be more verbose in our logging for tracing.
+StatusCode ExtProcFuzzHelper::randomHttpStatus() {
+  switch (provider_->ConsumeIntegralInRange<uint32_t>(0, 55)) {
+  case 0:
```

Comment:
It's not possible to use enum values here somehow?

Reply:
[FuzzedDataProvider.ConsumeEnum()](https://github.com/llvm/llvm-project/blob/main/compiler-rt/include/fuzzer/FuzzedDataProvider.h#L292) requires that enums start at 0, be contiguous, and have kMaxValue point to the largest value. For the gRPC status I think there is a cleaner way, but I'm open to suggestions for handling HttpStatus since they are non-contiguous.

Reply:
Maybe `PickValueInArray`, and then use the symbolic names of the enum values for the array. You can just log the picked value.